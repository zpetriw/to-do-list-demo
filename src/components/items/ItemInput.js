import React from 'react';
import { compose } from 'recompose';
import { withStyles } from 'material-ui/styles';
import PropTypes from 'prop-types';
import Button from 'material-ui/Button';
import TextField from 'material-ui/TextField';

const styles = theme => ({
    textField: {
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
        width: 200,
    },
    button: {
        margin: theme.spacing.unit,
    }
});

export const ItemInput = ({classes, textLabel, textValue, buttonLabel, onChange, onClick, inputRef}) => {
    return (
        <div>
            <TextField
            required
            id="required"
            label={textLabel}
            className={classes.textField}
            margin="normal"
            onChange={onChange}
            value={textValue}
            inputRef={inputRef}
            />
            <Button variant="raised" className={classes.button} onClick={onClick}>
                {buttonLabel}
            </Button>
        </div>
    );
};

ItemInput.propTypes = {
    classes: PropTypes.object.isRequired,
    textLabel: PropTypes.string.isRequired,
    textValue: PropTypes.string.isRequired,
    buttonLabel: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    inputRef: PropTypes.func,
};

export default compose(
    withStyles(styles)
)(ItemInput);