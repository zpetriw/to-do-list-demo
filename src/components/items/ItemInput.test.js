import React from 'react';
import expect from 'expect';
import {shallow, render} from 'enzyme';
import {ItemInput} from './ItemInput';
import Button from 'material-ui/Button';
import TextField from 'material-ui/TextField';
import renderer from 'react-test-renderer';

// Note: 
// toBe() means the two references must be for the same object.
// toEqual() does a deep equality check to see if two objects have the same keys.
// For primitives (int, string, ...) these functions will produce the same results.

describe('When ItemInput renders it', () => {

    let wrapper;
    let props;
    const onChangeInput = () => {};
    const onClickInput = () => {};

    beforeEach( () => {

        props = {
            classes: {    
                textField: "Text class",
                button: "Button class",
            },
            textLabel: "Text label",
            textValue: "Text Value",
            buttonLabel: "Button label",
            onChange: onChangeInput,
            onClick: onClickInput
        };
        
        wrapper = shallow(<ItemInput {...props}/>);
    });
    
    it('should render correctly', () => {
        const tree = renderer
            .create(<ItemInput {...props}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render <TextField> and <Button>', () => {
        expect(wrapper.find(TextField).length).toBe(1);
        expect(wrapper.find(Button).length).toBe(1);
    });

    it('should render <TextField> with the correct props', () => {
        expect(wrapper.find(TextField).props().label).toBe("Text label");
        expect(wrapper.find(TextField).props().className).toBe("Text class");
        expect(wrapper.find(TextField).props().onChange).toBe(onChangeInput);
    });

    it('should run onChangeInput when onChange is triggered', () => {
        expect();
    });

    it('should render <Button> with the correct props', () => {
        expect(wrapper.find(Button).props().className).toBe("Button class");
        expect(wrapper.find(Button).props().children).toBe("Button label");
    });

});