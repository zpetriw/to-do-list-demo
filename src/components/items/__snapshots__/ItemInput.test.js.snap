// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When ItemInput renders it should render correctly 1`] = `
<div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 Text class"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-11 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor="required"
    >
      Text label
      <span
        className="MuiFormLabel-asterisk-15"
      >
        â€‰*
      </span>
    </label>
    <div
      className="MuiInput-root-17 MuiInput-formControl-18 MuiInput-underline-21"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-25"
        defaultValue={undefined}
        disabled={false}
        id="required"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value="Text Value"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-46 MuiButton-root-31 MuiButton-raised-36 Button class"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-32"
    >
      Button label
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
</div>
`;
